Update splitting-a-subfolder-out-into-a-new-repository.md, adding com…

rifoerster opened the pull request:
…mand to create a repository out of a single subdirectory

The sub-directory will be the new root directory of the resulting repository

<!--
Thank you for contributing to this project! You must fill out the information below before we can review this pull request. By explaining why you're making a change (or linking to an issue) and what changes you've made, we can triage your pull request to the best possible team for review.
-->

### Why:
Adding the command for a common use case, where the split off sub-folder should become the root folder for the new repository.

Closes: #29639

<!-- If there's an existing issue for your change, please link to it above.
If there's _not_ an existing issue, please open one first to make it more likely that this update will be accepted: https://github.com/github/docs/issues/new/choose. -->

### What's being changed (if available, include any code snippets, screenshots, or gifs):
Adding a code snipped (shell code)

<!-- Let us know what you are changing. Share anything that could provide the most context.
If you made changes to the `content` directory, a table will populate in a comment below with links to the preview and current production articles. -->

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline.

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

rifoerster committed:
Update splitting-a-subfolder-out-into-a-new-repository.md, adding command to create a repository out of a single subdirectory 

The sub-directory will be the new root directory of the resulting repository

rifoerster committed:
Update splitting-a-subfolder-out-into-a-new-repository.md, added example command output

added example command output, as required by 
Content Lint Markdown / lint-content

rifoerster committed:
Update splitting-a-subfolder-out-into-a-new-repository.md

MD031, blanks-around-fences Fenced code blocks should be surrounded by blank lines

rifoerster committed:
Update splitting-a-subfolder-out-into-a-new-repository.md

 MD009, no-trailing-spaces Trailing spaces

subatoi commented on content/get-started/using-git/splitting-a-subfolder-out-into-a-new-repository.md:
```diff
@@ -61,6 +61,24 @@ If you create a new clone of the repository, you won't lose any of your Git hist
 
    The repository should now only contain the files that were in your subfolder(s).
 
+   If you want one specific subfolder to be the new root folder of the new repository, the following command can be used:
```
```suggestion
   If you want one specific subfolder to be the new root folder of the new repository, you can use the following command:
```

This is just a subtle amendment to use the active voice

subatoi commented on content/get-started/using-git/splitting-a-subfolder-out-into-a-new-repository.md:
````diff
@@ -61,6 +61,24 @@ If you create a new clone of the repository, you won't lose any of your Git hist
+   ```shell
+   $ git filter-repo --subdirectory-filter FOLDER-NAME
+   # Filter the specific branch by using a single sub-directory as the root for the new repository
+   > Parsed 82 commits
+   > New history written in 0.44 seconds; now repacking/cleaning...
+   > Repacking your repo and cleaning out old unneeded objects
+   > HEAD is now at 968d5a5 +: Print example / Notifier part 1
+   > Enumerating objects: 101, done.
+   > Counting objects: 100% (101/101), done.
+   > Delta compression using up to 6 threads
+   > Compressing objects: 100% (58/58), done.
+   > Writing objects: 100% (101/101), done.
+   > Total 101 (delta 44), reused 73 (delta 36), pack-reused 0
+   > Completely finished after 1.01 seconds.
````
I realise you're following the convention the article already has here from lines 58 and 59, but I'd recommend that we delete these and _also_ lines 58 and 59, as the output of this command may change at some point, and the commits in this example don't necessarily match lines 58/59.

```suggestion
```

subatoi reviewed (commented):
Thanks for this addition, @rifoerster! I agree it's worthwhile, but I've just made some small suggestions, so let me know what you think. If you agree, let's delete lines 58 and 59 in this PR, too.

rifoerster committed:
Update content/get-started/using-git/splitting-a-subfolder-out-into-a-new-repository.md

suggestion subatoi

Co-authored-by: Ben Ahmady <32935794+subatoi@users.noreply.github.com>

rifoerster committed:
Update content/get-started/using-git/splitting-a-subfolder-out-into-a-new-repository.md

suggestion subatoi

Co-authored-by: Ben Ahmady <32935794+subatoi@users.noreply.github.com>

rifoerster committed:
Update splitting-a-subfolder-out-into-a-new-repository.md

suggestion by subatoi
as command output will differ a lot depending on the repository

rifoerster commented:
I committed the suggestions. Now i only have to wait and see if the check will succeed. @subatoi

subatoi reviewed (approved):
Thanks for making those changes! I'll get this merged
